src/main/webapp/app/entities/legal-document.model.ts

```diff
@@ -0,0 +1,20 @@
+export enum LegalDocumentLanguage {
+    GERMAN = 'de',
+    ENGLISH = 'en',
+}
+
+export enum LegalDocumentType {
+    PRIVACY_STATEMENT = 'privacy-statement',
+    IMPRINT = 'imprint',
+}
+
+export class LegalDocument {
+    constructor(type: LegalDocumentType, language: LegalDocumentLanguage) {
+        this.type = type;
+        this.language = language;
+    }
+
+    language: LegalDocumentLanguage;
+    text: string;
+    type: LegalDocumentType;
```

Comment:
```suggestion
    language: LegalDocumentLanguage;
    text: string;
    type: LegalDocumentType;

    constructor(type: LegalDocumentType, language: LegalDocumentLanguage) {
        this.type = type;
        this.language = language;
    }
```

This is the convention I believe. Also, why is `text` not set in the constructor?

Reply:
we never set the text directly (at least not in production code), so I don't see much value in introducing this as a parameter here.